stub-idp/src/main/java/uk/gov/ida/stub/idp/resources/idp/PreRegistrationPageResource.java

```diff
@@ -0,0 +1,123 @@
+package uk.gov.ida.stub.idp.resources.idp;
+
+import uk.gov.ida.saml.core.domain.AuthnContext;
+import uk.gov.ida.stub.idp.Urls;
+import uk.gov.ida.stub.idp.cookies.CookieFactory;
+import uk.gov.ida.stub.idp.domain.DatabaseIdpUser;
+import uk.gov.ida.stub.idp.domain.SubmitButtonValue;
+import uk.gov.ida.stub.idp.exceptions.IncompleteRegistrationException;
+import uk.gov.ida.stub.idp.exceptions.InvalidDateException;
+import uk.gov.ida.stub.idp.exceptions.InvalidUsernameOrPasswordException;
+import uk.gov.ida.stub.idp.exceptions.UsernameAlreadyTakenException;
+import uk.gov.ida.stub.idp.repositories.Idp;
+import uk.gov.ida.stub.idp.repositories.IdpStubsRepository;
+import uk.gov.ida.stub.idp.services.IdpUserService;
+import uk.gov.ida.stub.idp.views.ErrorMessageType;
+import uk.gov.ida.stub.idp.views.PreRegistrationPageView;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+import java.util.Optional;
+
+import static uk.gov.ida.stub.idp.views.ErrorMessageType.INCOMPLETE_REGISTRATION;
+import static uk.gov.ida.stub.idp.views.ErrorMessageType.INVALID_DATE;
+import static uk.gov.ida.stub.idp.views.ErrorMessageType.INVALID_USERNAME_OR_PASSWORD;
+import static uk.gov.ida.stub.idp.views.ErrorMessageType.NO_ERROR;
+import static uk.gov.ida.stub.idp.views.ErrorMessageType.USERNAME_ALREADY_TAKEN;
+
+
+@Path(Urls.PRE_REGISTER_RESOURCE)
+@Produces(MediaType.TEXT_HTML)
+public class PreRegistrationPageResource {
```

Comment:
there is probably a better way to do this to re-use the exiting registration page without duplicating it

Reply:
I've re-used the existing registration page and resource now.